### PR TITLE
remove static analysis from CI

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: #1422

### CODE REVIEW

**Review by RickWinter — COMMENTED** [review]
How much time does this save on CI?  There is some risk that we miss catching bugs early.

**Comment by vhvb1989** [comment]
> 
> 
> How much time does this save on CI? There is some risk that we miss catching bugs early.

~14 min with no static code analysis:
https://dev.azure.com/azure-sdk/public/_build/results?buildId=695585&view=results

vs ~34 min with code analysis:
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=695584&view=results

The saving is 20 min per PR. 

>  There is some risk that we miss catching bugs early.

We haven't had any pipeline fail in the past due to this analysis.
And also, we can always add a comment to the PR as `/azp run cpp service-name` to trigger the liveTests for the PR and hence run this analysis. The benefit is that you can wait until PR is green and ready to merge to trigger this final check instead of having it per each commit

File: sdk/core/ci.yml
Comment (ahsonkhan) [inline sdk/core/ci.yml:36]: What's the purpose of this parameter?

**Review by ahsonkhan — APPROVED** [review]
Seems reasonable to me. We can monitor the number of live tests failure that happen because of this, and if it becomes too common, we can revert the change then, relatively easily.

File: sdk/core/ci.yml
Comment (vhvb1989) [reply sdk/core/ci.yml:36]: Enables liveTesting stage for a pipeline.

When it is not set, there are not liveTests run for that pipeline.  Then the value defines the regex that is used to run the tests like  `ctest -V -R azure-core.`

All the tests from azure-core are prefixed with `azure-core.` , so this regex would only run those tests.

For Storage, for instance, the regex is azure-storage. :)

**Pull request merged**
